elements/dash-tab-panel/dash-tab-panel.unittest.js

```diff
@@ -0,0 +1,60 @@
+const expect = chai.expect;
+
+describe('dash-tab-panel', function() {
+  before(dashElements.before());
+  after(dashElements.after());
+  beforeEach(function() {
+    this.container.innerHTML = `
+      <dash-tab-panel>
+        <header aria-role="tab" aria-controls="tab1">Tab 1</header>
+        <section id="tab1" aria-role="tabpanel">Content 1</section>
+        <header aria-role="tab" aria-controls="tab2">Tab 2</header>
+        <section id="tab2" aria-role="tabpanel">Content 2</section>
+        <header aria-role="tab" aria-controls="tab3">Tab 3</header>
+        <section id="tab3" aria-role="tabpanel">Content 3</section>
+      </dash-tab-panel>
+    `;
+    this.tabpanel = this.container.querySelector('dash-tab-panel');
+    this.tabs = Array.from(
+      this.container.querySelectorAll('[aria-role="tab"]')
+    );
+    this.panels = Array.from(
+      this.container.querySelectorAll('[aria-role="tabpanel"]')
+    );
+  });
+
+  it('should only have one visible panel initially',
+    dashElements.elementTest('dash-tab-panel', function() {
+      expect(this.panels.filter(p => !p.classList.contains('hidden')))
+        .to.have.lengthOf(1);
+    })
+  );
+
+  it('should have one selected tab initially',
+    dashElements.elementTest('dash-tab-panel', function() {
+      expect(this.tabs.filter(p => p.classList.contains('selected')))
+        .to.have.lengthOf(1);
+    })
+  );
+
+  it('should switch visibility when calling `selectTab()`',
```

Comment:
This is probably a longer discussion than one we want to have on this thread, but I was wondering if you might consider getting rid of the public `selectTab` method, and replacing it with a property. It would make the element more declarative and it would work better with libraries like React and Angular2 which favor a top down data flow model. I tried to explain my reasoning a bit here: http://robdodson.me/interoperable-custom-elements/#favoradeclarativeapioveranimperativeapi